### PR TITLE
Unbreak the unregister dialog after catalog changes

### PR DESCRIPTION
**Pull request by freben**
Screenshot 2020-06-08 at 16 43 24

### CODE REVIEW

**Comment by stefanalund** [comment]
The `github` prefix looks a bit funky.

**Comment by freben** [comment]
Yeah there is no custom renderer for the location - it's just shown verbatim

**Review by Nek — APPROVED** [review]

File: plugins/catalog/src/data/utils.tsx [hunk 1]
```diff
@@ -38,13 +39,12 @@ const createEditLink = (location: Location): string => {
   }
 };
 
-export function entityToComponent(
-  envelope: Entity,
-  location?: Location,
-): Component {
+export function entityToComponent(envelope: Entity): Component {
```
[the remark this answers is not in the document]
Comment (freben) [reply plugins/catalog/src/data/utils.tsx:41]: To be done! I'll do some more cleanup around this later

File: packages/app/src/components/Root/Root.tsx [hunk 2]
```diff
@@ -57,11 +57,14 @@ const SidebarLogo: FC<{}> = () => {
...
-        </Link>
-      </NavLink>
+      <Link
+        component={NavLink}
+        to="/"
+        underline="none"
+        className={classes.link}
+      >
+        {isOpen ? <LogoFull /> : <LogoIcon />}
+      </Link>
```
Comment (freben) [inline packages/app/src/components/Root/Root.tsx:67]: unrelated change - got tired of getting console warnings about nested `a` tags

File: plugins/catalog/src/api/CatalogClient.ts [hunk 3]
```diff
@@ -45,14 +45,27 @@ export class CatalogClient implements CatalogApi {
...
+        )
+        .join('&');
+    }
+    const response = await fetch(url);
+    if (!response.ok) {
+      const payload = await response.text();
+      throw new Error(
+        `Request failed with ${response.status} ${response.statusText}, ${payload}`,
+      );
+    }
```
Comment (freben) [inline plugins/catalog/src/api/CatalogClient.ts:67]: this change is about adding support for query filters, and handling non-200 responses properly
